### ISSUE TITLE
Add multi-thread support

### ISSUE THREAD

**Issue by lukeed** [lukeed/uvu#14]
Should be behind a CLI flag & should also be an opt-in – not everything is better/faster just because it's spread across more processes.

Having something simple like `uvu --parallel` is a nice, quick way to determine is it's a good fit for your project.
Similarly, `--parallel` should optionally accept a number of processes to fork into. This will also be limited by the number of machine cores. Omitting a number defaults to the host's core count (as seen by `os.cpus().length`).

This will be reserved for a major-version release – whether that's 1.0 or 2.0.
Definitely on the agenda, but it's a nice-to-have for 98% of projects. There are other, more pressing must-have's that much come first.

**Comment by OmgImAlexis** [comment]
Also maybe a way to mark a test as being able to be done in parallel or not?

**Comment by giuseppeg** [comment]
@lukeed have you thought about a viable implementation already? Would you be happy with something like this https://github.com/giuseppeg/uvu/commit/5974fe53ecd7fa1f1f779ed2b086f1269e00fd03 ?

**Comment by lukeed** [comment]
@giuseppeg Nice! I have a working pass at it now, using `worker_threads` which will require Node 10+ (8.x is current minimum, so it'd be a breaking change).

The idea is mostly the same, except that threads are cheaper to spawn and can set up a worker pool for mega number of test suites.

In addition to the breaking version requirement, I have this marked as "future" because I want to wait and see what other needs/features develop before locking us into a particular setup.

Until proven otherwise, I still don't see this as a "must-have ASAP" feature, and so it's just something to keep in back of our minds while accepting/building other aspects.

**Comment by meoyawn** [comment]
why go full multi-thread instead of going with simple asynchrony first.
I reckon it'd be much easier to implement, and remain lightweight.
What do you think?

```js
test.concurrent("name", async () => {
})
```

**Comment by dkstrong** [comment]
You can also tell if a function is async by looking at function.constructor.name. It would be really cool if in the in the test execution code (it think somewhere around here?: https://github.com/lukeed/uvu/blob/master/src/index.js#L138 ) it would look to see if it's async function or not. if it is an async function put it in a array of promises, then call Promise.all(promisesOfTests) at the end to allow all of the tests to run at the same time.
